src/main/java/edu/cornell/kfs/sys/service/impl/AwsSecretServiceImpl.java

```diff
@@ -0,0 +1,164 @@
+import com.google.gson.Gson;
+
+import edu.cornell.kfs.sys.jsonadapters.JsonDateSerializer;
+import edu.cornell.kfs.sys.service.AwsSecretService;
+
+public class AwsSecretServiceImpl  implements AwsSecretService{
+    private static final Logger LOG = LogManager.getLogger(AwsSecretServiceImpl.class);
+    
+    protected String awsRegion;
+    protected String kfsInstanceNameSpace;
+    protected String kfsSharedNamespace;
+    
+    @Override
+    public String getSingleStringValueFromAwsSecret(String awsKeyName, boolean useKfsInstanceNamespace) {
+        String fullAwsKey = buildFullAwsKeyName(awsKeyName, useKfsInstanceNamespace);
+        GetSecretValueRequest getSecretValueRequest = new GetSecretValueRequest().withSecretId(fullAwsKey);
+        GetSecretValueResult getSecretValueResult = null;
+        
+        AWSSecretsManager client = buildAWSSecretsManager();
+        try {
+            getSecretValueResult = client.getSecretValue(getSecretValueRequest);
+        } catch (DecryptionFailureException | InternalServiceErrorException | InvalidParameterException | 
+                InvalidRequestException | ResourceNotFoundException e) {
+            LOG.error("getSingleStringValueFromAwsSecret, had an error getting value for secret " + fullAwsKey, e);
+            throw new RuntimeException(e);
+        } finally {
+            client.shutdown();
+        }
+
+        return getSecretValueResult.getSecretString();
+    }
+    
+    @Override
+    public void updateSecretValue(String awsKeyName, boolean useKfsInstanceNamespace, String keyValue) {
+        String fullAwsKey = buildFullAwsKeyName(awsKeyName, useKfsInstanceNamespace);
+        UpdateSecretRequest updateSecretRequest = new UpdateSecretRequest ().withSecretId(fullAwsKey);
+        updateSecretRequest.setSecretString(keyValue);
+        
+        AWSSecretsManager client = buildAWSSecretsManager();
+        try {
+            perfornUpdate(updateSecretRequest, client);
+        } catch (EncryptionFailureException | InternalServiceErrorException | InvalidParameterException | LimitExceededException |
+                InvalidRequestException | ResourceNotFoundException | ResourceExistsException | MalformedPolicyDocumentException |
+                PreconditionNotMetException e) {
+            LOG.error("updateSecretValue, had an error setting value for secret " + fullAwsKey, e);
+            throw new RuntimeException(e);
+        } finally {
+            client.shutdown();
+        }
+    }
+
+    protected void perfornUpdate(UpdateSecretRequest updateSecretRequest, AWSSecretsManager client) {
+        boolean processed = false;
+        int tryCount = 0;
+        while(!processed) {
+            UpdateSecretResult result = client.updateSecret(updateSecretRequest);
+            if (result.getSdkHttpMetadata().getHttpStatusCode() == 200) {
+                processed = true;
+            } else {
+                tryCount ++;
+                if (tryCount <= 5) {
```

Comment:
Should the condition be "tryCount >= 5" instead?

Also, you don't need a space between the variable name and the "++"

Reply:
yes, thanks for catching